src/Diagnostics.ModelsAndUtils/Models/OperationContext.cs

```diff
@@ -25,30 +26,44 @@ public class OperationContext<TResource> where TResource : IResource
         /// </summary>
         public string EndTime { get; private set; }
 
+        /// <summary>
+        /// sets to false when detector should run as it is being called from external source (Azure portal, CLI ...)
+        /// sets to true when detector should run as it is being called from internal source (Applens ..)
+        /// </summary>
+        public bool IsInternalCall { get; private set; }
+
         /// <summary>
         /// sets to false when detector is called from external source (Azure portal, CLI ...)
         /// sets to true when detector is called from internal source (Applens ..)
         /// </summary>
-        public bool IsInternalCall { get; private set; }
+        public bool IsInternalClient { get; private set; }
```

Comment:
what is InternalClient? applens or asc?

Reply:
IsInternalClient means Applens really. ASC I added that 'forceInternal' flag for insights API, so it works a little differently, though I could probably improve that logic.

Reply:
Thats same as IsInternalCall, no?

Reply:
see my below comment and let me know if that makes more sense.